portfolio/src/main/webapp/blogScript.js

```diff
@@ -1,14 +1,23 @@
+const role = new URL(document.URL).searchParams.get('user');
+const id = new URL(document.URL).searchParams.get('id');
+
 window.addEventListener('DOMContentLoaded', (event) => {
-    getURL();
+  const user = document.getElementById('see-post');
+  const admin = document.getElementById('write-post');
+  if (role !== "admin" && id !== null) {
+    getID();
     getPost();
     getComments();
+    user.style.display = 'block';
+  }
+  else {
+    setAdmin(admin);
```

Comment:
Okay, so as you probably know, this isn't security in the slightest.

A secure implementation would require checking the user service on the server for each and every sensitive request sent to it. But we'd also have talked through how to implement a secure system prior to implementation :)